pkg/storage/storageclass.go

```diff
@@ -0,0 +1,301 @@
+)
+
+// ClassBuilder provides struct for storageclass object containing
+// connection to the cluster and the storageclass definitions.
+type ClassBuilder struct {
+	// Storageclass definition. Used to create the storageclass object.
+	Definition *storageV1.StorageClass
+	// Created storageclass object
+	Object *storageV1.StorageClass
+	// Used in functions that define or mutate storageclass definition. errorMsg is processed before the storageclass
+	// object is created.
+	errorMsg  string
+	apiClient *clients.Settings
+}
+
+// AdditionalOptions additional options for storageclass object.
+type AdditionalOptions func(builder *ClassBuilder) (*ClassBuilder, error)
+
+// NewClassBuilder creates a new instance of ClassBuilder.
+func NewClassBuilder(apiClient *clients.Settings, name, provisioner string) *ClassBuilder {
+	glog.V(100).Infof(
+		"Initializing new storageclass structure with the following params: "+
+			"name: %s, provisioner: %s", name, provisioner)
+
+	builder := ClassBuilder{
+		apiClient: apiClient,
+		Definition: &storageV1.StorageClass{
+			ObjectMeta: metaV1.ObjectMeta{
+				Name: name,
+			},
+			Provisioner: provisioner,
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the storageclass is empty")
+
+		builder.errorMsg = "storageclass 'name' cannot be empty"
+	}
+
+	if provisioner == "" {
+		glog.V(100).Infof("The provisioner of the storageclass is empty")
+
+		builder.errorMsg = "storageclass 'provisioner' cannot be empty"
+	}
+
+	return &builder
+}
+
+// WithReclaimPolicy adds a reclaimPolicy to the storageclass definition.
+func (builder *ClassBuilder) WithReclaimPolicy(
+	reclaimPolicy v1.PersistentVolumeReclaimPolicy) *ClassBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	if reclaimPolicy == "" {
+		glog.V(100).Infof("The reclaimPolicy of the storageclass is empty")
+
+		builder.errorMsg = "storageclass 'reclaimPolicy' cannot be empty"
+	}
+
+	if builder.errorMsg != "" {
+		return builder
+	}
+
+	builder.Definition.ReclaimPolicy = &reclaimPolicy
+
+	return builder
+}
+
+// WithVolumeBindingMode adds a volumeBindingMode to the storage class definition.
+func (builder *ClassBuilder) WithVolumeBindingMode(
+	bindingMode storageV1.VolumeBindingMode) *ClassBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	if bindingMode == "" {
+		glog.V(100).Infof("The bindingMode of the storageclass is empty")
+
+		builder.errorMsg = "storageclass 'bindingMode' cannot be empty"
+	}
+
+	if builder.errorMsg != "" {
+		return builder
+	}
+
+	builder.Definition.VolumeBindingMode = &bindingMode
+
+	return builder
+}
+
+// WithParameter adds a parameter to the storage class definition.
+func (builder *ClassBuilder) WithParameter(parameterKey, parameterValue string) *ClassBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	if parameterKey == "" {
+		glog.V(100).Infof("The parameterKey of the storageclass is empty")
```

Comment:
nit: there's no notion of parameterKey nor parameterValue for sc.
I'd rephrase to:
the "parameter's key" and "parameter's value" respectively